src/WebJobs.Script/Host/BlobLeaseManager.cs

```diff
@@ -123,62 +126,48 @@ private void ProcessLeaseTimerTick(object state)
                 }
             }
-            catch (StorageException exc)
+            else
             {
-                if (exc.RequestInformation.HttpStatusCode == 409)
+                // LockManager will handle various cases like 500s, 404s, retry if container doesn't exist, etc.
+                string proposedLeaseId = _instanceId;
+                this.LockHandle = await _lockManager.TryLockAsync(null, lockName, _instanceId, proposedLeaseId, _leaseTimeout, CancellationToken.None);
+                if (this.LockHandle == null)
                 {
-                    // If we did not have the lease already, a 409 indicates that another host had it. This is
-                    // normal and does not warrant any logging.
-
-                    if (HasLease)
-                    {
-                        // The lease was 'stolen'. Log details for debugging.
-                        string lastRenewalFormatted = _lastRenewal.ToString("yyyy-MM-ddTHH:mm:ss.FFFZ", CultureInfo.InvariantCulture);
-                        int millisecondsSinceLastSuccess = (int)(DateTime.UtcNow - _lastRenewal).TotalMilliseconds;
-                        int lastRenewalMilliseconds = (int)_lastRenewalLatency.TotalMilliseconds;
-                        ProcessLeaseError($"Another host has acquired the lease. The last successful renewal completed at {lastRenewalFormatted} ({millisecondsSinceLastSuccess} milliseconds ago) with a duration of {lastRenewalMilliseconds} milliseconds.");
-                    }
+                    // We didn't have the lease and failed to acquire it. Common if somebody else already has it.
+                    // This is normal and does not warrant any logging.
+                    return;
                 }
-                else if (exc.RequestInformation.HttpStatusCode >= 500)
-                {
-                    ProcessLeaseError($"Server error {exc.RequestInformation.HttpStatusMessage}.");
-                }
-                else if (exc.RequestInformation.HttpStatusCode == 404)
-                {
-                    // The blob or container do not exist, reset the lease information
-                    ResetLease();
 
-                    // Create the blob and retry
-                    _lockBlob = await GetLockBlobAsync(_lockBlob.ServiceClient, GetBlobName(_hostId));
-                    await AcquireOrRenewLeaseAsync();
-                }
-                else
-                {
-                    throw;
-                }
+                _lastRenewal = DateTime.UtcNow;
+                _lastRenewalLatency = _lastRenewal - requestStart;
+
+                string message = $"Host lock lease acquired by instance ID '{_instanceId}'.";
+                _traceWriter.Info(message);
+                _logger?.LogInformation(message);
+
+                // We've successfully acquired the lease, change the timer to use our renewal interval
+                SetTimerInterval(_renewalInterval);
             }
         }
 
```

Comment:
We still need it, but perhaps we could give it a better name. This is the "lockId" - ie, the name of what's being locked. 

---
In reply to: [123674089](https://github.com/Azure/azure-webjobs-sdk-script/pull/1615#discussion_r123674089) [](ancestors = 123674089)